### PR TITLE
[DOCS] Update concat and concat_ws documentation to point out unexpected behavior

### PR DESCRIPTION
**Pull request by watfordkcf**
Add documentation covering unexpected behavior of concat and concat_ws with respect to null values.

### What changes were proposed in this pull request?
Adds additional documentation to `concat` and `concat_ws`.

### Why are the changes needed?
The behavior of `concat` and `concat_ws` were unexpected w.r.t. null values and the documentation did not help make their behavior clear.

### Does this PR introduce _any_ user-facing change?
No.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala [hunk 1]
```diff
@@ -53,13 +53,17 @@ import org.apache.spark.unsafe.types.{ByteArray, UTF8String}
  */
 // scalastyle:off line.size.limit
 @ExpressionDescription(
-  usage = "_FUNC_(sep[, str | array(str)]+) - Returns the concatenation of the strings separated by `sep`.",
+  usage = "_FUNC_(sep[, str | array(str)]+) - Returns the concatenation of the strings separated by `sep`, skipping null values.",
```
Comment (yaooqinn) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala:56]: It's a null intolerant expression. Nulls are not skipped and the behavior is expected as null propagation. FYI, https://spark.apache.org/docs/latest/sql-ref-null-semantics.html#null-intolerant-expressions-.

**Review by yaooqinn — APPROVED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala (hunk 1 above)
Comment (yaooqinn) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala:56]: It looks like Pg got the wrong answer

**Comment by yaooqinn** [comment]
https://github.com/apache/spark/pull/42153/checks?check_run_id=15326884988 please follow the instructions here to enable the CI

**Comment by watfordkcf** [comment]
It shows as being enabled on my end, but I guess GitHub wasn't ready when I pushed the commit, so I squashed and force pushed.

**Comment by yaooqinn** [comment]
Would you mind retriggering the failed GitHub action jobs？

**Comment by watfordkcf** [comment]
GitHub would only let me retrigger the whole suite unfortunately.

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master/3.5/3/4

**Pull request closed without merging**